src/main/java/lavalink/client/io/jda/JdaLavalink.java

```diff
@@ -79,6 +79,10 @@ public JdaLink getExistingLink(Guild guild) {
     @NonNull
     public JDA getJda(int shardId) {
         if (jdaProvider == null) throw new IllegalStateException("JDA Provider not initialised!");
+
+        JDA result = jdaProvider.apply(shardId);
+        if (result == null) throw new IllegalStateException("JDA Provider returned null when trying to get JDA instance!");
```

Comment:
```suggestion
        if (result == null) throw new IllegalStateException("JDAProvider returned null for shard " + shardId);
```

Reply:
Wow I need to get better at describing errors, thanks, implementing